chef/cookbooks/cpe_screensaver/resources/cpe_screensaver.rb

```diff
@@ -39,30 +58,226 @@
+      ss_type = '/System/Library/Screen Savers/'
+    end
+
+    # Values for traditional built-in screensavers (quartz)
+    if [
+      'Arabesque',
+      'Word of the Day',
+    ].include? ss_module
+      ss_type = '/System/Library/Screen Savers/'
+      ss_num = 1
+      ss_ext = '.qtz'
+    end
+
+    # Values for iLifeSlideshows built-in screensavers
+    if [
+      '1-National Geographic',
+      '2-Aerial',
+      '3-Cosmos',
+      '4-Nature Patterns',
+    ].include? ss_module
+      ss_type = '/System/Library/Frameworks/ScreenSaver.framework/Resources/'
+      ss_module = 'iLifeSlideshows'
+    end
+
+    # Values for iLifeSlideshows built-in screensavers (custom folder)
+    if ss_module.include?('/')
+      ss_type = '/System/Library/Frameworks/ScreenSaver.framework/Resources/'
+      ss_module = 'iLifeSlideshows'
+    end
+
+    # Appends the com.apple.screensaver.Basic.ByHost payload to the profile hash
+    # if the conditional attribute is non nil. Sets the appropriate values for
+    # the MESSAGE style screensaver
+    if node['cpe_screensaver']['MESSAGE']
+      ss_type = '/System/Library/Frameworks/ScreenSaver.framework/Resources/'
+      ss_module = 'Computer Name'
+      ss_message = node['cpe_screensaver']['MESSAGE']
+      screensaver_profile['PayloadContent'].push(
+        'PayloadType' => 'com.apple.ManagedClient.preferences',
+        'PayloadVersion' => 1,
+        'PayloadIdentifier' => "#{prefix}.screensaver.Basic",
+        'PayloadUUID' => '8dd9a983-f59b-47c3-a408-ac55d287cbd4',
+        'PayloadEnabled' => true,
+        'PayloadDisplayName' => message,
+        'PayloadContent' => {
+          'com.apple.screensaver.Basic.ByHost' => {
+            'Set-Once' => [
+              {
+                'mcx_preference_settings' => {
+                  'MESSAGE' => ss_message,
+                },
+              },
+            ],
+          },
+        },
+      )
+    end
+
+    # Appends the com.apple.screensaver.ByHost payload to the profile hash
+    # with appropriate values for the following style screensavers;
+    # MESSAGE, built-in, 3rd party, custom folder
+    # Note we set the idleTime value her too...this is where the key is meant
+    # to be, if the profile is remove it will stick around.
+    screensaver_profile['PayloadContent'].push(
+      'PayloadType' => 'com.apple.ManagedClient.preferences',
+      'PayloadVersion' => 1,
+      'PayloadIdentifier' => "#{prefix}.screensaver.ByHost",
+      'PayloadUUID' => '01cb34f8-36f8-4fb6-babc-5ae3aa6c165b',
+      'PayloadEnabled' => true,
+      'PayloadDisplayName' => ss_module,
+      'PayloadContent' => {
+        'com.apple.screensaver.ByHost' => {
+          'Set-Once' => [
+            {
+              'mcx_preference_settings' => {
+                'idleTime' => node['cpe_screensaver']['idleTime'],
+                'moduleDict' => {
+                  'moduleName' => ss_module,
+                  'path' => ss_type + ss_module + ss_ext,
```

Comment:
For joining paths, you definitely want to use File.join (http://ruby-doc.org/core-2.2.0/File.html#method-c-join)